allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -1,20 +1,53 @@
-from allennlp.data.dataset import InstanceCollection
+from typing import Iterator
+
+from allennlp.data.dataset import Dataset, InstanceCollection, LazyDataset
 from allennlp.data.instance import Instance
 from allennlp.common import Params
 from allennlp.common.registrable import Registrable
 
 
 class DatasetReader(Registrable):
     """
-    A ``DatasetReader`` reads data from some location and constructs a :class:`Dataset`.  All
-    parameters necessary to read the data apart from the filepath should be passed to the
-    constructor of the ``DatasetReader``.
+    A ``DatasetReader`` reads data from some location and constructs an
+    :class:`InstanceCollection`.  All parameters necessary to read the data apart from the filepath
+    should be passed to the constructor of the ``DatasetReader``.
     """
+    def __init__(self, lazy: bool = False) -> None:
+        self._lazy = lazy
+
     def read(self, file_path: str) -> InstanceCollection:
         """
-        Actually reads some data from the `file_path` and returns a :class:`Dataset`.
+        Actually reads some data from the `file_path` and returns an :class:`InstanceCollection`.
+
+        The base class implementation passes this off to :func:`_read_instances`, and then uses the
```

Comment:
my brain parses "passes this off to" as "hands the result to", maybe just say "calls"